File: tests/test-async-lib.js
/**
 * Library with async functions for testing async type generation.
 */

async function fetchUser(id) {
  return { id, name: 'Alice', email: 'alice@example.com', active: true };
}

async function searchProducts(query, limit) {
  return {
    results: [{ id: 1, name: 'Widget', price: 9.99 }],
    total: 1,
    query,
  };
}

function formatPrice(amount, currency) {
  return { formatted: `${currency}${amount.toFixed(2)}`, amount, currency };
}

class ApiClient {
  async getProfile(userId) {
    return { userId, displayName: 'Bob', role: 'admin' };
  }

  async postComment(postId, text) {
    return { commentId: 42, postId, text, createdAt: '2026-01-01' };
  }

  getVersion() {
    return { version: '1.0.0', build: 123 };
  }
}

module.exports = { fetchUser, searchProducts, formatPrice, ApiClient };

File: tests/test_async_lib.py
"""Library with async functions for testing async type generation."""
import asyncio


async def fetch_user(user_id):
    return {"id": user_id, "name": "Alice", "email": "alice@example.com", "active": True}


async def search_products(query, limit):
    return {"results": [{"id": 1, "name": "Widget", "price": 9.99}], "total": 1, "query": query}


def format_price(amount, currency):
    return {"formatted": f"{currency}{amount:.2f}", "amount": amount, "currency": currency}


class ApiClient:
    async def get_profile(self, user_id):
        return {"user_id": user_id, "display_name": "Bob", "role": "admin"}

    async def post_comment(self, post_id, text):
        return {"comment_id": 42, "post_id": post_id, "text": text, "created_at": "2026-01-01"}

    def get_version(self):
        return {"version": "1.0.0", "build": 123}
